packages/markdown/loader.mjs

```diff
@@ -470,8 +479,11 @@ export default async function demoLoader() {
                 relativeModuleID,
               );
 
-              // the file has already been processed
-              if (addedModulesRelativeToModulePath.has(relativeModuleFilePath)) {
```

Comment:
Not 100% sure, but wouldn't it be solved if we just make `addedModulesRelativeToModulePath` work per-variant?

```diff
           demos[demoName].relativeModules = {};
         }
 
-        const addedModulesRelativeToModulePath = new Set();
         await Promise.all(
           Array.from(relativeModules.get(demoName)).map(async ([relativeModuleID, variants]) => {
             for (const variant of variants) {
+              const addedModulesRelativeToModulePath = new Set();
               let raw = '';
               const relativeModuleFilePath = path.join(
                 path.dirname(moduleFilepath),
```

Reply:
Actually, on a second look, that doesn't seem to be equivalent. We may need to 

```tsx
const addedModulesRelativeToModulePath = { TS: new Set(), JS: new Set() };
```